radixdlt-java/src/main/java/com/radixdlt/client/core/network/epics/SubmitAtomEpic.java

```diff
@@ -84,14 +84,19 @@ private Completable waitForConnection(RadixNode node) {
 						.subscribe(
 							() -> emitter.onNext(SubmitAtomReceivedAction.of(request.getUuid(), request.getAtom(), node)),
 							e -> {
-								emitter.onNext(
-									SubmitAtomStatusAction.fromStatusNotification(
-										request.getUuid(),
-										request.getAtom(),
-										node,
-										new AtomStatusNotification(AtomStatus.EVICTED_INVALID_ATOM, new JsonObject())
-									));
-								emitter.onNext(SubmitAtomCompleteAction.of(request.getUuid(), request.getAtom(), node));
+								if (e instanceof SubmitAtomException) {
+									SubmitAtomException submitAtomException = (SubmitAtomException) e;
+									emitter.onNext(
+										SubmitAtomStatusAction.fromStatusNotification(
+											request.getUuid(),
+											request.getAtom(),
+											node,
+											new AtomStatusNotification(AtomStatus.EVICTED_INVALID_ATOM, submitAtomException.getData())
+										));
+									emitter.onNext(SubmitAtomCompleteAction.of(request.getUuid(), request.getAtom(), node));
+								} else {
+									emitter.onError(e);
```

Comment:
From this code it seems `SubmitAtomException`'s are delivered as "success" and all other exceptions are breaking "errors". Would that expectation be clear from the context on the receiving end?

Reply:
Yes, this is seen in `RadixApplicationAPI.java:105` + `RadixApplicationAPI.java:946`